lib/syskit/roby_app/plugin.rb

```diff
@@ -942,6 +957,32 @@ def self.setup_rest_interface(app, rest_api)
                 require "syskit/roby_app/rest_api"
                 rest_api.mount REST_API => "/syskit"
             end
+
+            def rotate_logs(rotated_logs)
+                plan.find_tasks(Syskit::LoggerService).running.each do |task|
+                    rotated_logs[task.execution_agent.on] = task.rotate_log
+                end
+            end
+
+            def upload_rotated_logs(log_upload_command_queue, rotated_logs)
+                rotated_logs.each do |process_server_name, log_file_name|
+                    log_upload_command_queue << Upload.new(process_server_name, log_file_name)
+                end
+            end
+
+            def log_upload_main(log_upload_command_queue)
+                while (transfer = log_upload_command_queue.pop)
+                    transfer.apply
+                end
+            end
+
+            Upload = Struct.new(
+                :process_server_name, :file
+            ) do
```

Comment:
Use this syntax to avoid crossing the 90 char line limit. For short lines, put everything on the same line. I don't have the impression that this one would be too long (I might be wrong though ...). Here, it would be

```
Upload = Struct.new(:process_server_name, :file) do
   ... 
end
```